packages/next/client/performance-relayer.js

```diff
@@ -1,78 +1,54 @@
-function isTypeSupported(type) {
-  if (self.PerformanceObserver && PerformanceObserver.supportedEntryTypes) {
-    return PerformanceObserver.supportedEntryTypes.includes(type)
-  }
-  return false
-}
+import { getCLS, getFID, getFCP, getLCP, getTTFB } from 'web-vitals'
 
-export function observeLayoutShift(onPerfEntry) {
-  if (isTypeSupported('layout-shift')) {
-    let cumulativeScore = 0
-    const observer = new PerformanceObserver(list => {
-      for (const entry of list.getEntries()) {
-        // Only count layout shifts without recent user input.
-        if (!entry.hadRecentInput) {
-          cumulativeScore += entry.value
-        }
-      }
+export default onPerfEntry => {
+  // Measure Cumulative Layout Shift
+  getCLS(({ id, value }) =>
+    onPerfEntry({
```

Comment:
Suggestion: would it be possible to pass the `onPerfEntry` function directly as the callback? I'm seeing a lot of very similar looking code for each of these metrics, and it may be possible to just use a single function (or consolidate in some other way).

Reply:
Simplified it :)

The different names for each was why I was creating separate functions, but that's not necessary anymore